src/utils/openmc/tallies.py

```diff
@@ -1443,19 +835,61 @@ def __init__(self):
         self._tallies_file = ET.Element("tallies")
 
 
-    def add_tally(self, tally):
+    def add_tally(self, tally, merge=False):
 
         if not isinstance(tally, Tally):
             msg = 'Unable to add a non-Tally {0} to the TalliesFile'.format(tally)
             raise ValueError(msg)
 
-        self._tallies.append(tally)
+        if merge:
+            merged = False
+
+            # Look for a tally to merge with this one
+            for i, tally2 in enumerate(self._tallies):
+
+                # If a mergeable tally is found
+                if tally2.can_merge(tally):
+
+                    # Replace tally 2 with the merged tally
+                    merged_tally = tally2.merge(tally)
+                    self._tallies[i] = merged_tally
+                    merged = True
+
+            # If not mergeable tally was found, simply add this tally
+            if not merged:
+                self._tallies.append(tally)
+
+        else:
+            self._tallies.append(tally)
 
 
     def remove_tally(self, tally):
         self._tallies.remove(tally)
 
 
+    def merge_tallies(self):
+
+        for i, tally1 in enumerate(self._tallies):
+            for j, tally2 in enumerate(self._tallies):
+
+                # Do not merge the same tally with itself
+                if i == j:
+                    continue
```

Comment:
My point was that any cases where j <= i are superfluous because they have already been checked. You should still be able to do n-way merges if changing the second loop to iterate over `[i+1:]`. It would just make it a little cleaner because you would avoid the superfluous checks, and you wouldn't have to include the `if i == j:` block.